src/v/raft/append_entries_buffer.cc

```diff
@@ -0,0 +1,136 @@
+#include "raft/append_entries_buffer.h"
+
+#include "raft/consensus.h"
+#include "raft/types.h"
+
+#include <seastar/core/do_with.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/util/later.hh>
+#include <seastar/util/variant_utils.hh>
+
+#include <exception>
+#include <variant>
+#include <vector>
+namespace raft {
+
+/**
+ * This class allow to buffer arguments of a callers waiting for mutex and
+ * when mutex is eventually available execute all buffered calls without
+ * releasing mutex in beetween single calls.
+ */
+
+append_entries_buffer::append_entries_buffer(
+  consensus& c, size_t max_buffered_elements)
+  : _consensus(c)
+  , _max_buffered(max_buffered_elements) {}
+
+ss::future<append_entries_reply>
+append_entries_buffer::enqueue(append_entries_request&& r) {
+    return ss::with_gate(_gate, [this, r = std::move(r)]() mutable {
+        // we do not use a semaphore as we do not want to wait, waiting
```

Comment:
so is this therefore unbounded? what is the back pressure then?

Reply:
it is not unbounded, we throw an exception, we indeed do not block. Using semaphore would cause requests reordering.